### ISSUE TITLE
About eval.py

### ISSUE THREAD

**Issue by Ryunosuke-Yanagida** [ykykyk112/STIG#1]
Thank you for sharing the code related to this excellent research.

In the current code, running eval.py displays the results for clean, denoised, and noise. However, would it be possible to visualize the fake images corrected by STIG?

**Comment by ykykyk112** [comment]
Thank you for your attention @Ryunosuke-Yanagida .

In the **train.py**, we provide the visualization results during the training. You can also get the individual image by running **inference .py**. After running **inference.py**, images (png format) are saved at the folder, named inference. Note that you should correct your own root path in the line 31 of **inference.py**.

**Comment by Ryunosuke-Yanagida** [comment]
Thank you for your response despite your busy schedule.

When executing inference.py, should inference_data and inference_params be set via command-line arguments?
Additionally, could you provide details on any necessary path or command modifications when running it with the sample dataset?

**Comment by ykykyk112** [comment]
I'm sorry to make you uncomfortable with the imperfect code expression. I just checked the README.md, added the command line for inference your custom dataset, and fixed the code of the path configuration.

Set the same model arguments at the inference stage to load the model parameters successfully. All the results are saved in the {experiment_name} folder, including training and inference.

**Comment by Ryunosuke-Yanagida** [comment]
I am currently running the training with the provided sample dataset using the following command:
python train.py --size 256 --data sample --epoch 10 --batch_size 1 --lr 0.00008 --device 0 --dst sample

The inference command is executed as follows:
python inference.py --size 256 --inference_data C:\STIG\results\sample\eval\noise --device 0 --dst sample --inference_params C:\STIG\results\sample\parameters_9_epoch.pt
The code is running correctly, but I have a few questions:

For the inference_data parameter in the inference command, is it correct to use the eval\noise folder created during training?
When executing the inference command, the files in the generated inference folder appear in a random order.
The files created in the inference folder are exactly the same as the images in the inference_data path, and the corrected images processed by STIG are not being displayed. Could you kindly guide me on how to resolve this issue?
Thank you for your time and support.

**Comment by ykykyk112** [comment]
The sample data is just the subset of the entire training data for providing an examples. If you want to train for your own project, you should adopt more larger dataset (almost 10k~20k samples).

**Q.** For the inference_data parameter in the inference command, is it correct to use the eval\noise folder created during training?
**A.** Samples saved at 'eval/noise' folder is the training samples that are same with generated images at 'datasets/sample/fake'. So, if you want to inference other samples, e.g. not training samples, you need to describe the folder_path at command line.

**Q.** When executing the inference command, the files in the generated inference folder appear in a random order.
**A.** If you want to generate the results with a pre-defined order, you can modify 'RealFakeDataset' of the 'utils/dataset.py' and image save code lines in the 'InferenceModel' class of the 'utils/metric.py' as you want.

**Q.** The files created in the inference folder are exactly the same as the images in the inference_data path, and the corrected images processed by STIG are not being displayed.
**A.** As i mentioned above, the sample data is just 1% of the entire training data. So, it's impossible to correctly achieve to an optimal optimization point. As wrote in my paper, i used 8k~20k data samples for training each benchmarks.

Thank you for your deep attention.
